File: SelectionSortVisual/SelectionSortVisual-Java/src/main/java/com/mycompany/selectionsortvisual/SelectionSortVisual.java
package com.mycompany.selectionsortvisual;

import java.util.Arrays;

public class SelectionSortVisual {

    public static void selectionSort(int[] array) {
        int n = array.length;

        System.out.println("Vetor inicial: " + Arrays.toString(array));
        System.out.println("----------------------------------");

        for (int i = 0; i < n - 1; i++) {
            int minIndex = i;

            System.out.println("Iteracao " + (i + 1));
            System.out.println("Assumindo menor valor na posicao " + i + ": " + array[i]);

            for (int j = i + 1; j < n; j++) {
                System.out.println("Comparando " + array[j] + " com " + array[minIndex]);

                if (array[j] < array[minIndex]) {
                    minIndex = j;
                    System.out.println("Novo menor encontrado: " + array[minIndex] + " na posicao " + minIndex);
                }
            }

            if (minIndex != i) {
                System.out.println("Trocando " + array[i] + " pelo " + array[minIndex]);

                int temp = array[i];
                array[i] = array[minIndex];
                array[minIndex] = temp;
            } else {
                System.out.println("Nenhuma troca necessaria");
            }

            System.out.println("Estado atual do vetor: " + Arrays.toString(array));
            System.out.println("----------------------------------");
        }

        System.out.println("Vetor ordenado: " + Arrays.toString(array));
    }

    public static void main(String[] args) {
        int[] array = {64, 25, 12, 22, 11};
        selectionSort(array);
    }
}

File: SelectionSortVisual/SelectionSortVisual/SelectionSortVisual.py
def selection_sort(array):
    n = len(array)

    print("Vetor inicial:", array)
    print("----------------------------------")

    for i in range(n - 1):
        min_index = i

        print(f"Iteracao {i + 1}")
        print(f"Assumindo menor valor na posicao {i}: {array[i]}")

        for j in range(i + 1, n):
            print(f"Comparando {array[j]} com {array[min_index]}")

            if array[j] < array[min_index]:
                min_index = j
                print(f"Novo menor encontrado: {array[min_index]} na posicao {min_index}")

        if min_index != i:
            print(f"Trocando {array[i]} pelo {array[min_index]}")

            temp = array[i]
            array[i] = array[min_index]
            array[min_index] = temp
        else:
            print("Nenhuma troca necessaria")

        print("Estado atual do vetor:", array)
        print("----------------------------------")

    print("Vetor ordenado:", array)


def main():
    array = [64, 25, 12, 22, 11]
    selection_sort(array)


if __name__ == "__main__":
    main()